test/SparqlParserTest.cpp

```diff
@@ -1276,3 +1276,86 @@ TEST(ParserTest, Order) {
         ParseException);
   }
 }
+
+TEST(ParserTest, Group) {
+  {
+    ParsedQuery pq =
+        SparqlParser("SELECT ?x WHERE { ?x :myrel ?y } GROUP BY ?x").parse();
+    ASSERT_EQ(pq._groupByVariables.size(), 1);
+    ASSERT_EQ(pq._groupByVariables[0].name(), "?x");
+  }
+  {
+    // grouping by a variable
+    ParsedQuery pq =
+        SparqlParser("SELECT ?x WHERE { ?x :myrel ?y } GROUP BY ?y ?x").parse();
+    ASSERT_EQ(pq._groupByVariables.size(), 2);
+    ASSERT_EQ(pq._groupByVariables[0].name(), "?y");
+    ASSERT_EQ(pq._groupByVariables[1].name(), "?x");
+  }
+  {
+    // grouping by an expression
+    ParsedQuery pq =
+        SparqlParser("SELECT ?x WHERE { ?x :myrel ?y } GROUP BY (?x - ?y) ?x")
+            .parse();
+    ASSERT_EQ(pq._groupByVariables.size(), 2);
+    auto variant = pq._rootGraphPattern._children[1].variant_;
+    ASSERT_TRUE(holds_alternative<GraphPatternOperation::Bind>(variant));
+    auto helperBind = get<GraphPatternOperation::Bind>(variant);
+    ASSERT_EQ(helperBind._expression.getDescriptor(), "?x-?y");
+    ASSERT_EQ(pq._groupByVariables[0].name(), helperBind._target);
+    ASSERT_EQ(pq._groupByVariables[1].name(), "?x");
```

Comment:
Don't we already have an `IsBind`-Matcher for this? Otherwise we definitely write one for here!
(and of course for the repetitions below).

Reply:
I have already tried this once (but failed). The block I would extract would have been 1301-1304. The Problem then is that it should return the helper bind but the Asserts return some different type.

Maybe also extracting line 1305 can solve this. I'll look whether the whole block 1301-1305 looks usefull to me.

Reply:
Just let me know when you need help:)

Reply:
For most of the time I completely missunderstood what you wanted here.

Yes we do already have a Matcher `IsBind` that checks the expression descriptor and the variable. The difference here (in all but one case) is that the tests only care about the expression.

To try it I have added a new Matcher that only checks for the expression's descriptor.

Reply:
Or you could extend the matcher to optionally ignore or not ignore the variable:)